File: iosApp/iosApp/Presentation/Main/CafeManagement/CastList/CastListAction.swift
//
//  CastListAction.swift
//  ConCafe
//
//  Created by 홍희표 on 2026/07/05.
//

import Foundation

enum CastListAction {
    case clickBack
    case clickAddCast
    case changeSearchQuery(String)
    case clickCast(String)
    case clickCastSchedule(String)
    case clickDeleteCast(String)
    case confirmDeleteCast
    case dismissDeleteCastDialog
    case clickLoadMoreCasts
    case dismissInfoMessage
}

File: composeApp/src/commonMain/kotlin/com/hhp227/concafe/presentation/main/cafemanagement/castlist/CastListAction.kt
package com.hhp227.concafe.presentation.main.cafemanagement.castlist

sealed interface CastListAction {
    data object ClickBack : CastListAction
    data object ClickAddCast : CastListAction
    data class ChangeSearchQuery(val value: String) : CastListAction
    data class ClickCast(val castId: String) : CastListAction
    data class ClickCastSchedule(val castId: String) : CastListAction
    data class ClickDeleteCast(val castId: String) : CastListAction
    data object ConfirmDeleteCast : CastListAction
    data object DismissDeleteCastDialog : CastListAction
    data object ClickLoadMoreCasts : CastListAction
    data object DismissInfoMessage : CastListAction
}
